hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/evm/Evm46ValidationSuite.java

```diff
@@ -1489,6 +1498,34 @@ private HapiSpec internalCallToExistingSystemAccount800ResultsInSuccessNoop() {
                                 .hasTinyBars(changeFromSnapshot("initialBalance", 0)));
     }
 
+    @HapiTest
+    final HapiSpec testBalanceOfForSystemAccounts() {
+        final var contract = "BalanceChecker46Version";
+        final var BALANCE = 10L;
+        final var SYSTEM_ACCOUNT_BALANCE = 0L;
+        final HapiSpecOperation[] opsArray = new HapiSpecOperation[systemAccounts.size() * 2];
+        for (int i = 0; i < systemAccounts.size(); i++) {
+
+            // add contract call for all accounts in the list
+            opsArray[i] = contractCall(contract, BALANCE_OF, mirrorAddrWith(systemAccounts.get(i)))
+                    .hasKnownStatus(SUCCESS);
+
+            // add contract call local for all accounts in the list
+            opsArray[systemAccounts.size() + i] = contractCallLocal(
+                            contract, BALANCE_OF, mirrorAddrWith(systemAccounts.get(i)))
+                    .has(ContractFnResultAsserts.resultWith()
+                            .resultThruAbi(
+                                    getABIFor(FUNCTION, BALANCE_OF, contract),
+                                    ContractFnResultAsserts.isLiteralResult(
+                                            new Object[] {BigInteger.valueOf(SYSTEM_ACCOUNT_BALANCE)})))
+                    .hasAnswerOnlyPrecheck(OK);
```

Comment:
@Neeharika-Sompalli or @tinker-michaelj the API usage of this test seems different than the typical hapi test. Does this look alright?

Reply:
Ultimately a spec just executes an array of operations, so while unorthodox, this achieves the goal quite nicely. 😁